### ISSUE TITLE
getting NVDM run

### ISSUE THREAD

**Issue by benam2** [hugochan/KATE#26]
Any idea how to get NVDM works?
I need it for the comparison purpose to report its performance over different parameters than the ones reported in the paper.

Thanks~

**Comment by hugochan** [comment]
> Any idea how to get NVDM works?
> I need it for the comparison purpose to report its performance over different parameters than the ones reported in the paper.
> 
> Thanks~

You can use the open source implementation of NVDM [here](https://github.com/carpedm20/variational-text-tensorflow).

**Comment by benam2** [comment]
I bet you wish you get rid of my questions. Did the topics you shared in Table 2 in the paper was extracted from the whole 20 news data or only over the test data?

**Comment by hugochan** [comment]
> I bet you wish you get rid of my questions. Did the topics you shared in Table 2 in the paper was extracted from the whole 20 news data or only over the test data? And also the visualization case is over test data or the whole data?
> 
> Its gonna be my last question :))
> 
> I really thank you for taking time and answeing all the details :)


To extract the topics as in Table 2, for autoencoders, after training the model on the training set, each topic is represented by the 10 words (i.e., input neurons) with the strongest connection to that topic (i.e., hidden neuron). So you extract topics from the learned input-hidden weight matrix.

The document representation visualization is over the test set.

**Comment by benam2** [comment]
Thank you again, I got it for the visualization.

But for the table 2, you mean the extracted words reported is over the train data.
Am I correct? (Not training over the whole data, and then reporting the strongest connections with the neurons).

My point is that, why should we miss test data while we are training the model? if the purpose is not evaluation and just having good quality topics.

**Comment by hugochan** [comment]
> Thank you again, I got it for the visualization.
> 
> But for the table 2, you mean the extracted words reported is over the train data.
> Am I correct? (Not training over the whole data, and then reporting the strongest connections with the neurons).
> 
> My point is that, why should we miss test data while we are training the model? if the purpose is not evaluation and just having good quality topics.


Right. The model training was done over the training set in all my experiments since we needed to evaluate the model on the reserved test set. Yes,  if the purpose is not evaluation, you can train the model on the whole dataset.

**Comment by benam2** [comment]
Did you submit your work fist into arxive and then submitted to KDD or vise versa?
I appreciate if you share your experience?

**Comment by hugochan** [comment]
> Did you submit your work fist into arxive and then submitted to KDD or vise versa?
> I appreciate if you share your experience?

I first submitted it to KDD, and put it on arXiv after the author notification date. Different conferences have different policies.

**Comment by benam2** [comment]
You mean after acceptance date? right?
